test/e2e/annotations/secureupstream.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package annotations
+
+import (
+	"crypto/tls"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/parnurzeal/gorequest"
+	"k8s.io/ingress-nginx/test/e2e/framework"
+	"strings"
+)
+
+var _ = framework.IngressNginxDescribe("Annotations - SecureUpstream", func() {
+	f := framework.NewDefaultFramework("secureupstream")
+
+	BeforeEach(func() {
+		f.NewEchoDeployment()
+		f.NewGRPCFortuneTellerDeployment()
+	})
+
+	AfterEach(func() {
+	})
+
+	It("should set secure-verify-ca-secret with https backend", func() {
+		host := "secureupstream.foo.com"
+		nameSpace := f.IngressController.Namespace
+
+		_, err := framework.CreateIngressMASecret(
+			f.KubeClientSet,
+			host,
+			host,
+			nameSpace)
+		Expect(err).ToNot(HaveOccurred())
+
+		annotations := map[string]string{
+			"nginx.ingress.kubernetes.io/backend-protocol":        "HTTPS",
+			"nginx.ingress.kubernetes.io/secure-verify-ca-secret": host,
```

Comment:
The secret `host` is used for communication between client and ingress-nginx, can you use another secret here to make it realistic? (I actually expect the test to fail when you use a different secret, because the secret won't be available in the pod)

Reply:
@ElvinEfendi updated the PR with your suggestions, however I'm not sure how to check that the certificate is being proxied by NGINX. I have the tests just make sure that the configuration is generated, and we trust nginx to pass it to the upstream.